### PR TITLE
cloud: refactor unpublishing of resource links

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: api/cloud/oc_cloud_rd.c [hunk 1]
```diff
@@ -208,60 +208,60 @@ publish_published_resources(void *data)
...
-  if (data->code != OC_STATUS_DELETED)
-    goto error;
-  rd_link_free(&ctx->rd_delete_resources);
-error : {
-}
 }
+#endif /* OC_DEBUG */
 
 static void
-delete_resources(oc_cloud_context_t *ctx, bool all)
```
Comment (jkralik) [inline api/cloud/oc_cloud_rd.c:227]: All need to be there - it means all resources at cloud must be unpublished
If all is not set then only selected resources need to be deleted.

Comment (Danielius1922) [reply api/cloud/oc_cloud_rd.c:227]: All was never used.

Comment (jkralik) [reply api/cloud/oc_cloud_rd.c:227]: I know but API counts with it. Maybe put there comment about that device can unpublish all resources without ins

Comment (Danielius1922) [reply api/cloud/oc_cloud_rd.c:227]: API? But it's a static helper function, surely that cannot be regarded as an API function.

File: api/cloud/oc_cloud_rd.c [hunk 2]
```diff
@@ -208,60 +208,60 @@ publish_published_resources(void *data)
...
     return;
   }
-  if (ctx->rd_delete_resources) {
-    rd_delete(ctx->cloud_ep, ctx->rd_delete_resources, ctx->device,
-              delete_resources_handler, LOW_QOS, ctx);
+
+#ifdef OC_DEBUG
+  oc_response_handler_t handler = delete_resources_handler;
+#else  /* !OC_DEBUG */
+  oc_response_handler_t handler = NULL;
```
Comment (jkralik) [inline api/cloud/oc_cloud_rd.c:240]: `handler` need to be always set. otherwise segfault can occurs.

**Review by jkralik — APPROVED** [review]

**Pull request merged**